### DISCUSSION TITLE
why the dependencies did not update

### DISCUSSION THREAD

**Discussion by jiangxiaoqiang** [pdm-project/pdm#2457] [Q&A] [answered]
I have already using the pdm update command to update the dependencies and the log show the update success:

```
> pdm update
🔒 Lock successful
Synchronizing working set with resolved packages: 0 to add, 11 to update, 0 to remove

  ✔ Update setuptools 68.2.2 -> 69.0.2 successful
  ✔ Update wheel 0.41.3 -> 0.42.0 successful
  ✔ Update exceptiongroup 1.1.3 -> 1.2.0 successful
  ✔ Update certifi 2023.7.22 -> 2023.11.17 successful
  ✔ Update pytest-xdist 3.4.0 -> 3.5.0 successful
  ✔ Update idna 3.4 -> 3.6 successful
  ✔ Update rdpywheel 0.1.32 -> 0.1.33 successful
  ✔ Update parse 1.19.1 -> 1.20.0 successful
  ✔ Update pytest-rerunfailures 12.0 -> 13.0 successful
  ✔ Update pygments 2.16.1 -> 2.17.2 successful
  ✔ Update seleniumbase 4.21.4 -> 4.21.6 successful

```

but the project still use the legacy version of the dependencies, should I refresh the dependencies? what should I do to make it use the newest dependencies?

the dependencies path may look like:

`/Users/xiaoqiangjiang/source/dolphin/visa/venv/lib/python3.10/site-packages/rdpywheel/auth/JtwHandler.py
`

<img width="611" alt="image" src="https://github.com/pdm-project/pdm/assets/4521375/c6ad17cd-96cc-468d-b885-5eaae06c2bd6">

the rdpywheel newest version is 0.1.33. This is the pdm info:

```

> pdm info
PDM version:
  2.10.4
Python Interpreter:
  /Users/xiaoqiangjiang/source/dolphin/visa/.venv/bin/python (3.10)
Project Root:
  /Users/xiaoqiangjiang/source/dolphin/visa
Local Packages:

```

and this is the pdm info --env:

```
> pdm info --env
{
  "implementation_name": "cpython",
  "implementation_version": "3.10.11",
  "os_name": "posix",
  "platform_machine": "arm64",
  "platform_release": "22.4.0",
  "platform_system": "Darwin",
  "platform_version": "Darwin Kernel Version 22.4.0: Mon Mar  6 20:59:28 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T6000",
  "python_full_version": "3.10.11",
  "platform_python_implementation": "CPython",
  "python_version": "3.10",
  "sys_platform": "darwin"
}


```

it look like pycharm did not use the pdm dependencies.

**Comment by pawamoy** [comment]
Can you show the output of `pdm info` and `pdm info --env` please?

**Reply by jiangxiaoqiang** [reply]
I have update the question to append the `pdm info` and `pdm info --env`

**Reply by pawamoy** [reply, marked as answer]
Thanks. `pdm info` shows `/Users/xiaoqiangjiang/source/dolphin/visa/.venv`, while the line above your screenshot shows `/Users/xiaoqiangjiang/source/dolphin/visa/venv`: `venv` instead of `.venv`. These are two different virtual environments. Reconciliate them and the issue should disappear :slightly_smiling_face:
